examples/step-101/step-101.cc

```diff
@@ -0,0 +1,650 @@
+              {
+                for (const unsigned int j : fe_values.dof_indices())
+                  {
+                    local_stiffness(i, j) += fe_values.shape_grad(i, q) *
+                                             fe_values.shape_grad(j, q) *
+                                             fe_values.JxW(q);
+                  }
+                local_rhs(i) += rhs_function.value(point) *
+                                fe_values.shape_value(i, q) * fe_values.JxW(q);
+              }
+          }
+
+        // Now we loop over faces. If a face borders a cell with FE_Nothing
+        // (i.e., a cell outside the domain), we apply shifted boundary
+        // conditions.
+        for (const auto &face : GeometryInfo<dim>::face_indices())
+          {
+            // Skip faces that border cells with active finite elements
+            if (cell->neighbor(face)->active_fe_index() ==
+                ActiveFEIndex::lagrange)
+              continue;
+
+            surface_fe_values.reinit(cell, face);
+
+            // Get the neighboring cell for level set evaluation
+            auto neighbour = cell->neighbor(face);
+            typename DoFHandler<dim>::cell_iterator neighbours_lvl_set_cell(
+              &neighbour->get_triangulation(),
+              neighbour->level(),
+              neighbour->index(),
+              &level_set_dof_handler);
+
+            neighbours_lvl_set_cell->get_dof_indices(level_set_dof_indices);
+            level_set.extract_subvector_to(level_set_dof_indices.begin(),
+                                           level_set_dof_indices.end(),
+                                           dof_values_level_set.begin());
+
+            // Loop over quadrature points on the face
+            for (const unsigned int q :
+                 surface_fe_values.quadrature_point_indices())
+              {
+                const Point<dim> &point = surface_fe_values.quadrature_point(q);
+
+                // @sect5{Computing the shift}
+                // For a point on the mesh boundary, we find the closest point
+                // on the true boundary Γ. For a sphere, this is simply the
+                // point projected onto the unit circle/sphere.
+                const Point<dim> closest_boundary_point =
+                  point * (1. / point.norm());
+
+                // Future work: Computing shifts from level set is being
+                // developed in PR #18680. This would allow for more general
+                // geometries.
+
+                // Transform the shifted point to the reference cell coordinates
+                const Point<dim> unit_shifted_point =
+                  surface_fe_values.get_mapping().transform_real_to_unit_cell(
+                    cell, closest_boundary_point);
+
+                const Point<dim> real_shifted_point =
+                  surface_fe_values.get_mapping().transform_unit_to_real_cell(
+                    cell, unit_shifted_point);
```

Comment:
Doesn't that give you back `closes_boundary_point`?  You transform `closest_boundary_point` from real space to unit space and then, transform that point from unit space to real space.